subsys/net/ip/net_if.c

```diff
@@ -410,6 +410,13 @@ static void join_mcast_solicit_node(struct net_if *iface,
 	struct in6_addr addr;
 	int ret;
 
+#if defined(CONFIG_NET_L2_BT)
+	/* No need to join solicited-node, RFC 7668 ch 3.2.2 */
```

Comment:
It's introducing bearer specific logic in net_if, not great.

Looks like the mcast logic might need a rework so L2 could request it relevantly, maybe (just quick thought)

Reply:
Probably changing mcast logic is not very trivial thing to do.

If we want to avoid any mention of actual bearer in net_if.c, then we could have a new config option(s) that would be set by BT subsystem.
So instead of having

```
#if defined(CONFIG_NET_L2_BT)
	/* No need to join solicited-node, RFC 7668 ch 3.2.2 */
	if (iface->l2 == &NET_L2_GET_NAME(BLUETOOTH)) {
		return;
	}
#endif
```

we could do

```
#if defined(CONFIG_NET_IPV6_SKIP_JOIN_SOLICIT_NODE)
	if (iface->l2 == CONFIG_NET_IPV6_SKIP_JOIN_SOLICIT_NODE_L2) {
		return;
	}
#endif
```

And then have BT Kconfig file to set CONFIG_NET_IPV6_SKIP_JOIN_SOLICIT_NODE and CONFIG_NET_IPV6_SKIP_JOIN_SOLICIT_NODE_L2 appropriately. The CONFIG_NET_IPV6_SKIP_JOIN_SOLICIT_NODE_L2 could be a hidden option as user would not need to see this.

Reply:
Maybe an simpler way: 
- creating l2 flags  (some enum), >= 0
- l2->enable() would return < 0 in case of error, or 0>= 0 -> the l2 flag

And one of that flag could be MCAST support: if so, join mcast.

Reply:
Yes, that would be even simpler. We could go with route then.